modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java

```diff
@@ -127,54 +127,77 @@ public void testNullEntry() throws Exception {
         }
     }
 
-    @Test
-    public void testJSONPointerFunction() throws Exception {
-        ContentFeatureSource fs = dataStore.getFeatureSource(tname("jsontest"));
-        FilterFactory ff = dataStore.getFilterFactory();
+    private void testPointerEquals(ContentFeatureSource fs, FilterFactory ff, String column, String literalValue,
+                             double comparisonValue)
+            throws IOException {
         Function pointer =
-                ff.function("jsonPointer", ff.property("jsonColumn"), ff.literal("/weight"));
-        Literal literal = ff.literal(0.001);
+                ff.function("jsonPointer", ff.property(column), ff.literal(literalValue));
+        Literal literal = ff.literal(comparisonValue);
         Filter filter = ff.equals(pointer, literal);
-        Query q = new Query(tname("jsontest"), filter);
+        Query q = new Query(tname(column), filter);
         FeatureCollection collection = fs.getFeatures(q);
         try (FeatureIterator it = collection.features()) {
             int size = 0;
             while (it.hasNext()) {
                 Feature f = it.next();
-                assertEquals(pointer.evaluate(f), 0.001f);
+                assertEquals(pointer.evaluate(f), comparisonValue);
                 size++;
             }
             assertEquals(1, size);
         }
     }
 
     @Test
-    public void testJSONPointerFunctionWithArray() throws Exception {
+    public void testJSONPointerFunction() throws Exception {
         ContentFeatureSource fs = dataStore.getFeatureSource(tname("jsontest"));
         FilterFactory ff = dataStore.getFilterFactory();
+        testPointerEquals(fs, ff,"jsonColumn","/weight",0.001);
```

Comment:
I believe a simpler refactor would have been to extract a function that only takes the name of the property, everything else is actually the same, could have stayed in the delegate method. Just thinkng out loud, but I don't see other calls to testPointerEquals.

Reply:
Agreed and fixed.